src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -74,44 +72,43 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
       dmi: DataModelInstanceQueryResponse,
       requiredPropsArray: Seq[String]): ProjectedDataModelInstance = {
     val dmiProperties = dmi.properties.getOrElse(Map())
-    ProjectedDataModelInstance(
+    val res = ProjectedDataModelInstance(
       externalId = dmi.properties
         .flatMap(_.get("externalId"))
-        .flatMap(_.asString)
+        .map(_.asInstanceOf[StringProperty].value)
         .getOrElse(
           throw new CdfSparkException("Can't read data model instance, `externalId` is missing.")),
       properties = requiredPropsArray.map { name: String =>
```

Comment:
Are these properties required to exist in the result, as null or some, or are they actually required to be populated? Seems this code accepts null entries for the required properties, which was a little puzzling to me.

Reply:
required means, the selected properties i believe, name is just deceiving.